integrations/google_ai/src/haystack_integrations/components/generators/google_ai/chat/gemini.py

```diff
@@ -336,11 +340,26 @@ def _get_stream_response(
         :param streaming_callback: The handler for the streaming response.
         :returns: The extracted response with the content of all streaming chunks.
         """
-        responses = []
+        replies: Union[List[str], List[ChatMessage]] = []
```

Comment:
This is technically not correct, the list can be etherogeneous. This would be more correct.
```suggestion
        replies: List[Union[str, ChatMessage]] = []
```